### PR TITLE
feat: Support non-string parameters. Closes #1236

### LINKED ISSUE #1236: support non-string types in trigger parameter sources
**Is your feature request related to a problem? Please describe.**
If a destination field is not a string (ex: k8s Deployment replicas), then k8s trigger operations will fail.

The issue seems to be that https://github.com/argoproj/argo-events/blob/master/sensors/triggers/params.go#L236 getValueByKey(...) returns a string

Example
Use a service account that can create Deployments
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Sensor
metadata:
  name: webhook
spec:
  template:
    serviceAccountName: k8s-sa
  dependencies:
    - name: test-dep
      eventSourceName: webhook
      eventName: example
  triggers:
    - template:
        name: webhook-workflow-trigger
        k8s:
          group: apps
          version: v1
          resource: deployments
          operation: create
          source:
            resource:
              apiVersion: apps/v1
              kind: Deployment
              metadata:
                name: test-deployment
                labels:
                  app: whalesay
              spec:
                replicas: 1
                selector:
                  matchLabels:
                    app: whalesay
                template:
                    containers:
                    - name: cowsay
                      image: docker/whalesay:latest
                      command: [cowsay]
                      args: ["THIS_WILL_BE_REPLACED"]
          parameters:
            - src:
                dependencyName: test-dep
                dataKey: body.message
              dest: spec.template.containers.0.args.0
            - src:
                dependencyName: test-dep
                dataKey: body.replicas
              dest: spec.replicas
```

send the following event
```bash
curl -d '{"message":"this is my first webhook", "replicas": 2}' -H "Content-Type: application/json" -X POST http://localhost:12000/example
```

You'll see the following error in the sensor logs
```bash
{"level":"info","ts":1622402963.8126473,"logger":"argo-events.sensor","caller":"standard-k8s/standar-k8s.go:171","msg":"creating the object...","sensorName":"webhook","triggerName":"webhook-workflow-trigger","triggerType":"Kubernetes"}
{"level":"error","ts":1622402963.8473272,"logger":"argo-events.sensor","caller":"sensors/listener.go:271","msg":"failed to execute a trigger","sensorName":"webhook","error":"failed to execute trigger: timed out waiting for the condition: Deployment in version \"v1\" cannot be handled as a Deployment: v1.Deployment.Spec: v1.DeploymentSpec.Replicas: readUint32: unexpected character: �, error found in #10 byte of ...|eplicas\":\"2\",\"select|..., bigger context ...|t\",\"namespace\":\"argo-events\"},\"spec\":{\"replicas\":\"2\",\"selector\":{\"matchLabels\":{\"app\":\"whalesay\"}},\"|...","errorVerbose":"timed out waiting for the condition: Deployment in version \"v1\" cannot be handled as a Deployment: v1.Deployment.Spec: v1.DeploymentSpec.Replicas: readUint32: unexpected character: �, error found in #10 byte of ...|eplicas\":\"2\",\"select|..., bigger context ...|t\",\"namespace\":\"argo-events\"},\"spec\":{\"replicas\":\"2\",\"selector\":{\"matchLabels\":{\"app\":\"whalesay\"}},\"|...\nfailed to execute trigger\ngithub.com/argoproj/argo-events/sensors.(*SensorContext).triggerOne\n\t/home/runner/work/argo-events/argo-events/sensors/listener.go:328\ngithub.com/argoproj/argo-events/sensors.(*SensorContext).triggerActions\n\t/home/runner/work/argo-events/argo-events/sensors/listener.go:269\ngithub.com/argoproj/argo-events/sensors.(*SensorContext).listenEvents.func1.3\n\t/home/runner/work/argo-events/argo-events/sensors/listener.go:181\nruntime.goexit\n\t/opt/hostedtoolcache/go/1.14.15/x64/src/runtime/asm_amd64.s:1373","triggerName":"webhook-workflow-trigger","triggeredBy":["test-dep"],"triggeredByEvents":["31396262316664352d333237632d346232632d396434362d383730396335386662383038"],"stacktrace":"github.com/argoproj/argo-events/sensors.(*SensorContext).triggerActions\n\t/home/runner/work/argo-events/argo-events/sensors/listener.go:271\ngithub.com/argoproj/argo-events/sensors.(*SensorContext).listenEvents.func1.3\n\t/home/runner/work/argo-events/argo-events/sensors/listener.go:181"}
```

**Describe the solution you'd like**
Add an optional dataType field (defaulting to string) to the src to help resolve the parameters to a particular type supported by gjson. 

```yaml
            - src:
                dependencyName: test-dep
                dataKey: body.replicas
                dataType: int
              dest: spec.replicas
```

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by AalokAhluwalia**
Checklist:

* [X] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

**Comment by AalokAhluwalia** [comment]
Addresses #1236

File: pkg/apis/sensor/v1alpha1/types.go [hunk 1]
```diff
@@ -743,15 +743,20 @@ type TriggerParameterSource struct {
 	// To access an array value use the index as the key. The dot and wildcard characters can be escaped with '\\'.
 	// See https://github.com/tidwall/gjson#path-syntax for more information on how to use this.
 	DataKey string `json:"dataKey,omitempty" protobuf:"bytes,4,opt,name=dataKey"`
+	// UseRawDataValue indicates if the value in an event at data key should be used without converting to string.
+	// When true, a number, boolean, json or string parameter may be extracted. When the field is unspecified, or explicitly
+	// false, the behavior is to turn the extracted field into a string or json. (e.g. when set to true, the parameter
+	// 123 will resolve to the numerical type, but when false, or not provided, the string "123" will be resolved)
+	UseRawDataValue bool `json:"useRawDataValue,omitempty" protobuf:"bytes,5,opt,name=useRawDataValue"`
```
Comment (whynowy) [inline pkg/apis/sensor/v1alpha1/types.go:750]: 1. Please add it to the end of the struct - do not change the order of original properties.

2. If it not specified, consider it as the `string` (reverting https://github.com/argoproj/argo-events/pull/2273), in this case, if expecting a JSON, just set it to `true`.

Comment (AalokAhluwalia) [reply pkg/apis/sensor/v1alpha1/types.go]: 👍 made the change. let me know if the wording of this comment can be improved.

**Review by whynowy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**